### DISCUSSION TITLE
optional first `|`

### DISCUSSION THREAD

**Discussion by scarf005** [lark-parser/lark#1184] [Ideas]
it would be nice to be able to use
```
NUMBER_TYPE : 
    | INTEGER_TYPE
    | UNSIGNED_TYPE
    | FLOAT_TYPE
```
over current syntax.
```
NUMBER_TYPE : INTEGER_TYPE
    | UNSIGNED_TYPE
    | FLOAT_TYPE
```

i wonder if there's any technological difficulties for supporting optional first pipe symbol?

**Comment by MegaIng** [comment]
If you do this in a single line, it doesn't mean the same, i.e. it also accepts the empty expression. (see #1077) I could have sworn there also is a discussion somewhere about a different multi-line syntax where what you are proposing was one of the suggestions, but I can't find it.

**Reply by erezsh** [reply]
Do you mean this one? https://github.com/lark-parser/lark/issues/155

**Reply by scarf005** [reply]
well, this is certainly a confusing behavior and i can't find out anything about [docs](https://lark-parser.readthedocs.io/en/latest/grammar.html).
i think empty expression should be more explicit.

**Reply by MegaIng** [reply]
@erezsh Yes

@scarf005 It's just a consequence of the way the grammar syntax works, no special casing is happening.

**Reply by scarf005** [reply]
@MegaIng well then the way the grammar syntax works should be more explicitly documented, because i could not find out what would happen if the rule is empty. (try ctrl-f `empty` on docs)
